### ISSUE TITLE
onChisel should return an instance of the player who's currently using it.

### ISSUE THREAD

**Issue by CptSpaceToaster** [TheCricket/Chisel-2#297]
As per [issue 15](https://github.com/TehNut/MoreChisels/issues/15) from [MoreChisels](https://github.com/TehNut/MoreChisels)

The chisel interface we provide can't do some of the cool things that are being discussed.  Because the player instance isn't readily available, we can't make chisels that require experience to use, or chisel's that use Botania Mana, etc.

[Source](https://github.com/TheCricket/Chisel-2/blob/4264a87546eebe13bfc4e461716c8d2cc9a2b148/src/main/java/com/cricketcraft/chisel/api/IChiselItem.java#L39)

If we follow semantic-versioning, we might want to consider bumping up a larger increment, versus a patch if this is included.

**Comment by tterrag1098** [comment]
The player instance isn't passed because the player instance *can't* be passed all the time. Using a chisel does not require a player to be present (i.e. auto chisel).

**Comment by CptSpaceToaster** [comment]
We could send null in that instance?

**Comment by tterrag1098** [comment]
I suppose, but this will be a API breaking change no matter what.

**Comment by CptSpaceToaster** [comment]
Yup... you're correct... I don't think there's another way we can add this change without messing with the API... we're going to break things.

Should we hold off on that then?  I'd hate to say "We aren't going to change" out of fear of breaking stuff, but on the other hand... maybe we should wait a bit till we have more some more features.

**Comment by TheCricket** [comment]
I am going to propose a change

**Comment by CptSpaceToaster** [comment]
Doesn't look terrible... We'll have to be clear that this will break chisel compatibility with a lot of mods that already had it... so we should make sure that everything is in place before we release.
